combat-trainer.lic

```diff
@@ -2648,41 +2648,44 @@ class TrainerProcess
       reset_ability(game_state, 'Meraud')
       return
     end
-    if /meraud/i =~ bput('commune sense', 'roundtime', 'Meraud', 'Meraud\'s')
+    if /meraud/i =~ DRC.bput('commune sense', 'roundtime', 'Meraud', 'Meraud\'s')
       reset_ability(game_state, 'Meraud', 60)
       game_state.blessed_room = true
       return
     end
     waitrt?
     @equipment_manager.stow_weapon(game_state.weapon_name)
-    fput "get #{@water_holder} from my #{@theurgy_supply_container}"
-    if /that/i =~ bput("sprinkle #{@water_holder} on #{checkname}", 'You sprinkle', 'Sprinkle what?', 'Sprinkle that')
-      fput "put #{@water_holder} in my #{@theurgy_supply_container}"
+    unless DRCI.exists?(@water_holder, @theurgy_supply_container,)
+      DRC.message("Cannot get #{@water_holder}, skipping Meraud commune")
       @equipment_manager.wield_weapon?(game_state.weapon_name, game_state.weapon_skill)
       @training_abilities.delete('Meraud')
       return
     end
-    bput "put #{@water_holder} in my #{@theurgy_supply_container}", 'You put', 'What were you referring to'
-    fput "get #{@flint_lighter}", 'You get a', 'You are already'
-    if bput("get incense from my #{@theurgy_supply_container}", 'You get', 'You are already', 'referring to') == 'referring to'
-      fput "stow #{@flint_lighter}", 'You put your', 'Stow what'
+    unless DRCTH.has_holy_water?(@theurgy_supply_container, @water_holder)
+      DRC.message("Cannot get holy water, skipping Meraud commune")
       @equipment_manager.wield_weapon?(game_state.weapon_name, game_state.weapon_skill)
       @training_abilities.delete('Meraud')
       return
     end
-    fput 'light my incense with flint' # get messaging
-    pause
-    waitrt?
-    fput "wave incense at #{checkname}"
-    fput 'snuff incense'
-    bput("put incense in my #{@theurgy_supply_container}", 'You put', 'What were you referring to')
-    fput "stow #{@flint_lighter}", 'You put your', 'Stow what'
+    unless DRCTH.sprinkle_holy_water(@theurgy_supply_container, @water_holder, checkname)
+      DRC.message("Cannot sprinkle holy water, skipping Meraud commune")
+      @equipment_manager.wield_weapon?(game_state.weapon_name, game_state.weapon_skill)
+      @training_abilities.delete('Meraud')
+      return
+    end
+    unless DRCTH.wave_incense?(@theurgy_supply_container, @flint_lighter, checkname)
+      DRC.message("Cannot get incense to wave, skipping Meraud commune")
```

Comment:
There's an opportunity to combine all of this cleanup logic, something like:

```suggestion
unless DRCTH.sprinkle_holy_water?(@theurgy_supply_container, @water_holder, checkname) && DRCTH.wave_incense?(@theurgy_supply_container, @flint_lighter, checkname)
      DRC.message("Cannot sprinkle incense  and wave incense, skipping Meraud commune")
      @equipment_manager.wield_weapon?(game_state.weapon_name, game_state.weapon_skill)
      @training_abilities.delete('Meraud')
      return
  end
```

What do you think about that?